### PR TITLE
feat: dkp high for Kubecost, Centralized Kubecost, Kubernetes Dashboard

### PR DESCRIPTION
**Pull request by sonofgibs**
**What problem does this PR solve?**:
Update app configs to use DKP High PriorityClass for monitoring apps

**Which issue(s) does this PR fix?**:

https://d2iq.atlassian.net/browse/D2IQ-96663

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

File: services/centralized-kubecost/0.34.0/defaults/cm.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ metadata:
 data:
   values.yaml: |
     ---
+    priorityClassName: "dkp-high-priority"
```
Comment (gracedo) [inline services/centralized-kubecost/0.34.0/defaults/cm.yaml:9]: (applies to the below kubecost values as well) we use an umbrella chart that brings in upstream kubecost ("cost-analyzer" helm chart) as a subchart -- so all the upstream values are nested under `cost-analyzer` (line 10 in this diff here).

The upstream helm chart values are here: https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/values.yaml there are currently 2 places where priorityClassName can be set, but I am seeing now that these are both components that are not enabled by default. I'm going to actually look at contributing a PR upstream to add support to set priority class on all the deployments. Let's hold off on the 2 kubecost apps here until I get that in!

File: services/nvidia-gpu-operator/22.9.1/defaults/cm.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ metadata:
   namespace: ${releaseNamespace}
 data:
   values.yaml: |-
+    priorityClassName: "dkp-high-priority"
```
Comment (gracedo) [inline services/nvidia-gpu-operator/22.9.1/defaults/cm.yaml:8]: here's the helm chart values: https://github.com/NVIDIA/gpu-operator/blob/master/deployments/gpu-operator/values.yaml I see that they are already defaulting to using `system-node-critical`, which I think is probably the right thing to do here (I don't think we should override them) so let's actually not make any changes here! I'll make a note in the doc about this

```suggestion
```

**Review by gracedo — COMMENTED** [review]
Let's remove the override for nvidia, and also the kubecost apps for now (until I get an upstream PR in)

File: services/centralized-kubecost/0.34.0/defaults/cm.yaml (hunk 1 above)
Comment (gracedo) [reply services/centralized-kubecost/0.34.0/defaults/cm.yaml:9]: https://github.com/kubecost/cost-analyzer-helm-chart/blob/1723242ea6744b2d20cdda6dea9a4a59ffe22faa/cost-analyzer/values.yaml#L487-L490

Took a closer look at kubecost as I went in to make the changes -- they do actually support creating a priority class and using it in some (but not all) deployments. We can provide our own priority class name and it'll get used:

```
priority:
  enabled: true
  name: dkp-high-priority
```

I will still look into seeing whether I should open an upstream pr to allow setting it for all deployments not just the main kubecost ones but we can set this field now without waiting! This should be set in both kubecost and centralized-kubecost

File: services/kubecost/0.34.0/defaults/cm.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ data:
           enabled: true
         grafana:
           enabled: true
+        priorityClassName: "dkp-high-priority"
```
Comment (gracedo) [inline services/kubecost/0.34.0/defaults/cm.yaml:20]:
```suggestion
```

kubecost pulls in prometheus and grafana subcharts, we'll want to set priority classes for those deployments as well. For prometheus, [server](https://github.com/kubecost/cost-analyzer-helm-chart/blob/1723242ea6744b2d20cdda6dea9a4a59ffe22faa/cost-analyzer/charts/prometheus/values.yaml#L551) and [alertmanager](https://github.com/kubecost/cost-analyzer-helm-chart/blob/1723242ea6744b2d20cdda6dea9a4a59ffe22faa/cost-analyzer/charts/prometheus/values.yaml#L51). Doesn't look like they support setting priority classes for grafana yet, I'll look at adding it there too.

File: services/centralized-kubecost/0.34.0/defaults/cm.yaml (hunk 1 above)
Comment (sonofgibs) [reply services/centralized-kubecost/0.34.0/defaults/cm.yaml]: Good catch! Updates have been made

File: hack/validate-default-apps-versions.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-latestKommanderVersion=$(find "services/kommander" -maxdepth 1 -type d -regextype sed -regex '.*/[0-9]\+.[0-9]\+.[0-9]\+' -printf "%f\n" | sort --version-sort | tail -1)
+latestKommanderVersion=$(gfind "services/kommander" -maxdepth 1 -type d -regextype sed -regex '.*/[0-9]\+.[0-9]\+.[0-9]\+' -printf "%f\n" | sort --version-sort | tail -1)
```
Comment (cbuto) [inline hack/validate-default-apps-versions.sh:4]: the image used in CI won't have `gfind` unfortunately, which is why pre-commit is failing. 

you can get pre-commit working locally with:
1. `brew install findutils`
2. then add this in your `.zshrc` file:
```
PATH=$(brew --prefix)/opt/findutils/libexec/gnubin:$PATH
```

then source it in your shell and it should be good to go!

**Comment by gracedo** [comment]
closing, these are being taken care of in different PRs!

**Pull request closed without merging**